### PR TITLE
test: [Snaps E2E] Create E2E test for UI links

### LINKED ISSUE #22425: [Snaps E2E] Create test for links in custom UI
Test Links in custom UI (see snaps 2021)

### PR DESCRIPTION
**Pull request by bowensanders**
## **Description**

Create E2E test for UI links
## **Related issues**

Fixes: #22425

## **Manual testing steps**

1. This is an automated E2E test

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/snaps/test-snap-uilinks.spec.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        failOnConsoleError: false,
+        title: this.test.fullTitle(),
+      },
+      async ({ driver }) => {
+        await unlockWallet(driver);
+
+        // navigate to test snaps page and connect to dialog snap
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-uilinks.spec.js:22]: Does this need to be a separate test?

File: test/e2e/snaps/test-snap-uilinks.spec.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+  defaultGanacheOptions,
+  withFixtures,
+  unlockWallet,
+  WINDOW_TITLES,
+} = require('../helpers');
+const FixtureBuilder = require('../fixture-builder');
+const { TEST_SNAPS_WEBSITE_URL } = require('./enums');
+
+describe('Test Snap Dialog', function () {
+  it('test all three snap_dialog types', async function () {
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-uilinks.spec.js:11]: If this needs to be a separate test we should update all the of this

**Review by FrederikBolding — CHANGES_REQUESTED** [review]

File: test/e2e/snaps/test-snap-uilinks.spec.js (hunk 1 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-uilinks.spec.js:22]: Our original discussion on the topic landed on creating a links test seperately, future-proofing for other possible cases where links could happen, making seperate  ITs if necessary, but this one, at this point, only checks for a link in one of the three dialog types (where it has already been added to the example snap.)

File: test/e2e/snaps/test-snap-uilinks.spec.js (hunk 2 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-uilinks.spec.js]: I have made changes in the latest commit to reflect what is actually happening.

File: test/e2e/snaps/test-snap-uilinks.spec.js (hunk 1 above)
Comment (bowensanders) [reply test/e2e/snaps/test-snap-uilinks.spec.js:22]: It also leaves this test open to testing other UI features if necessary

File: test/e2e/snaps/test-snap-uilinks.spec.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        });
+
+        // wait for the link to be provided
+        await driver.waitForSelector({
+          text: 'snaps.metamask.io',
+          tag: 'b',
+        });
+
+        // wait for and click back button
+        await driver.waitForSelector({
```
Comment (FrederikBolding) [inline test/e2e/snaps/test-snap-uilinks.spec.js:98]: Should we not test that it opens the window?

Comment (FrederikBolding) [reply test/e2e/snaps/test-snap-uilinks.spec.js:98]: Yep!

Comment (bowensanders) [reply test/e2e/snaps/test-snap-uilinks.spec.js:98]: Added in the latest commit. It required adding hosts to the privacy snapshot but it works.

**Review by FrederikBolding — APPROVED** [review]

**Review by GuillaumeRx — APPROVED** [review]

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**